fontforge/python.c

```diff
@@ -4367,121 +4428,218 @@ struct flaglist strokeflags[] = {
     FLAGLIST_EMPTY /* Sentinel */
 };
 
-static int Stroke_Parse(StrokeInfo *si, PyObject *args) {
-    const char *str, *cap="butt", *join="round";
-    double width=0, minor=1, angle=0;
-    int c, j, f;
+#define STROKE_OPTKEYS "removeinternal", "removeexternal", "extrema", "simplify", "accuracy", "joinlimit", "extendcap", "jlrelative", "ecrelative", "removeoverlap"
+#define STROKE_OPTFORMAT "$ppppdddpps"
+#define STROKE_OPTARGS &si->removeinternal, &si->removeexternal, &si->extrema, &si->simplify, &si->accuracy_target, &si->joinlimit, &si->extendcap, &si->jlrelative, &si->ecrelative, &rostring
+
+static char *strokekey_circ[]
+    = { "type", "width", "cap", "join", "angle", STROKE_OPTKEYS, NULL };
+static char *strokebkey_circ[]
+    = { "type", "width", "cap", "join", "flags", NULL };
+static char *strokekey_ellip[]
+    = { "type", "width", "minorwidth", "angle", "cap", "join", STROKE_OPTKEYS, NULL };
+static char *strokebkey_ellip[]
+    = { "type", "width", "minorwidth", "angle", "cap", "join", "flags", NULL };
+static char *strokekey_rect[]
+    = { "type", "width", "height", "angle", "cap", "join", STROKE_OPTKEYS, NULL };
+static char *strokebkey_rect[]
+    = { "type", "width", "height", "angle", "flags", NULL };
+static char *strokekey_conv[]
+    = { "type", "contour", "angle", "cap", "join", STROKE_OPTKEYS, NULL };
+static char *strokebkey_conv[]
+    = { "type", "contour", "flags", NULL };
+
+static int Stroke_Parse(StrokeInfo *si, PyObject *args, PyObject *keywds) {
+    char *str, *type, *cap="nib", *join="nib", *rostring="layer";
+    int c, j, f, r, toknum;
     PyObject *flagtuple=NULL;
-    PyObject *poly=NULL;
-    int argcnt;
+    PyObject *nib=NULL;
 
     if ( !PySequence_Check(args) ) {
 	PyErr_Format(PyExc_TypeError, "Expected a sequence");
-return( -1 );
+	return( -1 );
     }
-    argcnt = PySequence_Size(args);
-    if ( argcnt==0 ) {
+    if ( PySequence_Size(args)==0 ) {
 	PyErr_Format(PyExc_TypeError, "Expected a name of a pen type");
-return( -1 );
+	return( -1 );
     }
+
+    InitializeStrokeInfo(si);
+    si->radius *= 2;
```

Comment:
Why is this doubled? Because the inputs expects a diameter?

Reply:
The API has parameters for width and height (or major and minor axis). So I double here for the default values and halve down below. 

In practice I don't think this does anything because the width api doesn't have a default and the default minorradius is 0.